File: CapstoneProject/SocialMediaPlatform/Frontend/social-media/src/app/services/profile.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Profile } from '../modals/profile';

@Injectable({
  providedIn: 'root'
})
export class ProfileService {

  constructor(private httpClient: HttpClient) { }

  getUserProfile(username: string) {
    console.log("Get Profile API called!")
    return this.httpClient.get<Profile>(`http://localhost:8080/api/profiles/${username}`)
  }

  updateProfile(username: string, profile: any) {
    console.log("Profile Service: Update Profile Method called!")
    return this.httpClient.put<Profile>(`http://localhost:8080/api/profiles/update/${username}`, profile)
  }

}

File: CapstoneProject/SocialMediaPlatform/Backend/SocialMediaApp/src/main/java/com/social/service/ProfileService.java
package com.social.service;

import org.springframework.http.ResponseEntity;

import com.social.dao.Profile;
import com.social.error.GlobalExceptionHandler;

public interface ProfileService {

	ResponseEntity<Profile> findByUserUsername(String username);

	ResponseEntity<Profile> update(String username, Profile profile) throws GlobalExceptionHandler;

}
